compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFusePackIntoForLoop.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "mlir/Dialect/SCF/Transforms/TileUsingInterface.h"
+#include "mlir/IR/Iterators.h"
+#include "mlir/Pass/Pass.h"
+
+#define DEBUG_TYPE "iree-amdaie-fuse-pack-into-for"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+class AMDAIEFusePackIntoForLoopPass
+    : public impl::AMDAIEFusePackIntoForLoopBase<
+          AMDAIEFusePackIntoForLoopPass> {
+ public:
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<scf::SCFDialect>();
+  }
+
+  AMDAIEFusePackIntoForLoopPass() = default;
+  AMDAIEFusePackIntoForLoopPass(const AMDAIEFusePackIntoForLoopPass &pass){};
+  void runOnOperation() override;
+};
+
+void AMDAIEFusePackIntoForLoopPass::runOnOperation() {
+  MLIRContext *context = &getContext();
+  func::FuncOp funcOp = getOperation();
+  IRRewriter rewriter(context);
+
+  // Walk through the graph in post order and find the pack op.
+  TilingInterface packOp;
+  funcOp->walk<WalkOrder::PostOrder, ReverseIterator>([&](TilingInterface op) {
+    if (isa<linalg::FillOp>(op)) {
+      packOp = op;
```

Comment:
This is looking for a fill op not a pack op?

Reply:
Thanks for catching this bug. This should be a tensor::PackOp.